### PR TITLE
fixes #1948 -- MD5 on FIPS systems

### LINKED ISSUE #1948: Use sha256 for account ids
md5 is an artifact from days long gone - so much so that it isn't supported at all by web crypto.

I'm working on a [node.js (and browser) clone of letsencrypt](https://git.daplie.com/Daplie/node-greenlock) that works on all platforms - including OS X and Windows (complete) and the browser (incomplete) and I don't know what every possible security consideration of an account id might be, but I'd like to stay 100% compatible and I'd also rather be safe than sorry.

### PR DESCRIPTION
**Pull request by JoshMcCullough**
## Pull Request Checklist

- [x] If the change being made is to a [distributed component](https://certbot.eff.org/docs/contributing.html#code-components-and-layout), edit the `master` section of `certbot/CHANGELOG.md` to include a description of the change being made.
- [x] Include your name in `AUTHORS.md` if you like.

This is the fix recommended in #1948. I am _not_ a Python developer so this needs a decent review, but the unit and integration tests pass, at least.

### CODE REVIEW

**Comment by bmw** [comment]
It looks like some of our automated tests are failing with these changes. Let me know if you'd like help fixing them.

**Comment by JoshMcCullough** [comment]
Yes, please.

**Comment by bmw** [comment]
The errors to fix are https://travis-ci.com/certbot/certbot/jobs/278395565#L579 and https://travis-ci.com/certbot/certbot/jobs/278395566#L577.

For the latter, you can just append `  # type: ignore` to the end of the line with `usedforsecurity`.

**Comment by JoshMcCullough** [comment]
@bmw fixed!

File: AUTHORS.md [hunk 1]
```diff
@@ -265,3 +265,4 @@ Authors
 * [YourDaddyIsHere](https://github.com/YourDaddyIsHere)
 * [Zach Shepherd](https://github.com/zjs)
 * [陈三](https://github.com/chenxsan)
+* [Josh McCullough](https://github.com/JoshMcCullough)
```
[the remark this answers is not in the document]
Comment (JoshMcCullough) [reply AUTHORS.md]: Funny, I looked at it to see if that was the case but my brain said <NOT SORTED>. :)

**Pull request merged**